sec_certs_page/common/bindings.py

```diff
@@ -0,0 +1,510 @@
+    """
+    try:
+        datetime.datetime.fromisoformat(timestamp)
+        return True
+    except ValueError:
+        return False
+
+
+def verify_url(url: str) -> bool:
+    """ Verifies URL validity
+
+    Args:
+        url (str): URL to validate
+
+    Returns:
+        bool: Whether the given URL is valid
+    """
+    try:
+        result = urllib.parse.urlparse(url)
+        return all([result.scheme, result.netloc])
+    except ValueError:
+        return False
+
+
+def verify_jwt(data: dict[str, Union[str, object]]) -> bool:
+    """ Verifies the given object jwt validity
+
+    Args:
+        data (dict[str, Union[str, object]]): object to validate
+
+    Returns:
+        bool: Whether the JWT in the data object is valid
+    """
+    jwt_token = data['JWT']
+    del data["JWT"]
+    encoded_jwt = jwt.encode(
+        data, key=current_app.config["BINDINGS_PUBLIC_KEY"], algorithm="RS256")
+    if not jwt_token:
+        click.echo("JWT token is missing")
+        return False
+    if encoded_jwt != jwt_token:
+        return False
+    return True
+
+
+async def download_file(file_url: str, file_path: str, sha256: str = "", verbose: bool = False) -> None:
+    """ Download a file from the given URL
+
+    Args:
+        file_url (str): URL of the file to download
+        file_path (str): Path to where to download the file
+        sha256 (str, optional): Expected hash of the downloaded file. Defaults to "".
+        verbose (bool, optional): Whether to click.echo out extra information. Defaults to False.
+    """
+    if verbose:
+        click.echo(f"Downloading {file_url}...")
+    response = requests.get(file_url, timeout=1000)
+    if sha256:
+        file_sha256 = hashlib.sha256(response.content).digest().hex()
+        if file_sha256 != sha256:
+            click.echo(
+                f"Metadata SHA256 verification failed for file {file_path.split('/')[-1]}")
+    try:
+        with open(file_path, "wb") as f:
+            f.write(response.content)
+    except:
+        click.echo(f"Cant open url from file {file_url}")
+    if verbose:
+        click.echo(f"Downloaded {file_url}.")
+
+
+async def download_binding_files(url: str, output_dir: str, file_ext: str = ".json", verbose: bool = False) -> None:
+    """ Download all binding files from a given url.
+
+    Args:
+        url (str): URL to download binding files from
+        output_dir (str): Path to write binding files to
+        file_ext (str): Optional file extension of binding files to be downloaded
+        verbose (bool, optional): Whether to click.echo out extra information. Defaults to False.
+    """
+    if verbose:
+        click.echo(f"Downloading bindings files from {url}")
+
+    if url.startswith("https://github.com"):
+        tasks = []
+        repo_url = url.split("/tree")[0]
+        dir_path = url.split("/main")[1]
+        api_url = github_url_to_api(repo_url) + "/contents" + dir_path
+
+        response = requests.get(api_url)
```

Comment:
We really need to be authenticated to Github in order for this to work, the rate limit is quite steep. Some 60 requests per few hours or so. I think our batch processing, even if only done daily or weekly could go over that with enough bindings.